python/ray/serve/examples/tutorials/sklearn+composition.rst

```diff
@@ -0,0 +1,344 @@
+
+    import requests
+    
+    sample_request_input = {
+        "sepal length": 1.2,
+        "sepal width": 1.0,
+        "petal length": 1.1,
+        "petal width": 0.9,
+    }
+    response = requests.get(
+        "http://localhost:8000/logistic_regressor", json=sample_request_input
+    )
+    response.text
+
+
+
+
+.. parsed-literal::
+
+    'setosa'
+
+
+
+.. parsed-literal::
+
+    [2m[36m(pid=37038)[0m Worker: received flask request with data {'sepal length': 1.2, 'sepal width': 1.0, 'petal length': 1.1, 'petal width': 0.9}
+
+
+Part 3: Deploy multiple models as a pipeline
+--------------------------------------------
+
+Often time we need to deploy multiple model to form a pipeline. In this
+tutorial, we will deploy an *ensemble* pipeline that compose two models
+and aggregate their responses.
+
+::
+
+             -> Logistic Regression -->
+            /                          \
+   Request ->                           -> Ensemble -> Response
+            \                          /
+             ->   SVM Classifier   -->
+
+To begin, let’s train both classifiers and save them.
+
+.. code:: ipython3
+
+    from sklearn.linear_model import LogisticRegression
+    from sklearn.svm import SVC
+    
+    models = {"logistic_regression": LogisticRegression(), "svm_classifier": SVC()}
+    for name, model in models.items():
+        model.fit(train_x, train_y)
+        with open("/tmp/iris_{}.pkl".format(name), "wb") as f:
+            pickle.dump(model, f)
+
+With the models saved to disk, let’s define our deployment class and
+deploy them with Serve’s low level deployment API. We will enable the
+sklearn model to accept *Python* input.
+
+.. code:: ipython3
+
+    class SKLearnModel:
+        def __init__(self, model_path, label_path="/tmp/iris_labels.json"):
+            with open(model_path, "rb") as f:
+                self.model = pickle.load(f)
+            with open(label_path) as f:
+                self.label_list = json.load(f)
+    
+        def __call__(self, flask_request, *, payload=None):
+            if serve.context.web:
+                payload = flask_request.json
+            else:
+                payload = payload
+    
+            input_vector = [
+                payload["sepal length"],
+                payload["sepal width"],
+                payload["petal length"],
+                payload["petal width"],
+            ]
+            prediction = self.model.predict([input_vector])[0]
+            human_name = self.label_list[prediction]
+            return human_name
+
+Serve’s endpoints are logical names for the service. It can be
+associated with several backends. Each backend correspond to a function
+or a class.
+
+.. code:: ipython3
+
+    serve.create_endpoint(endpoint_name="logistic", route="/logistic")
+    serve.create_endpoint(endpoint_name="svm", route="/smv")
+
+.. code:: ipython3
+
+    serve.create_backend(SKLearnModel, "logistic:v1", "/tmp/iris_logistic_regression.pkl")
```

Comment:
Can you explain what the last argument means? (It will be great if you can write code in this way).
```python3
serve.create_backend(
    SKLearnModel,                       # serve backend
    "logistic:v1",                      # logical name
    "/tmp/iris_logistic_regression.pkl" # argument
)
```